File: examples/python/send_web.py
"""Send an SMS via the Korean web gateway. Requires pre-registered sender."""
import os
import sys

from textory_client import TextoryClient, TextoryError


def main() -> int:
    sender = os.environ.get("TEXTORY_SENDER")
    if not sender:
        print("Set TEXTORY_SENDER=+821098765432 (a registered sender number).", file=sys.stderr)
        return 1

    tx = TextoryClient()
    try:
        result = tx.send_web(
            sender=sender,
            recipients=[{"phoneNumber": "+821012345678"}],
            contents="배송이 시작되었습니다. 감사합니다.",
        )
    except TextoryError as err:
        print(f"❌ {err.status} {err.code}: {err.message}", file=sys.stderr)
        return 1

    print(result)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: examples/nodejs/send-web.js
// Send an SMS through Textory's Korean web gateway. Sender must be pre-registered.
//   export TEXTORY_SENDER=+821098765432
//   node send-web.js

import { TextoryClient } from './client.js';

const apiKey = process.env.TEXTORY_API_KEY;
const sender = process.env.TEXTORY_SENDER;
if (!apiKey || !sender) {
  console.error('Set TEXTORY_API_KEY and TEXTORY_SENDER.');
  process.exit(1);
}

const tx = new TextoryClient(apiKey);

try {
  const result = await tx.sendWeb({
    sender,
    recipients: [{ phoneNumber: '+821012345678' }],
    contents: '배송이 시작되었습니다. 감사합니다.',
  });

  console.log('✅ Sent:', JSON.stringify(result, null, 2));
} catch (err) {
  if (err.response) {
    console.error(`❌ ${err.response.status}:`, JSON.stringify(err.response.data, null, 2));
  } else {
    console.error('❌ Network error:', err.message);
  }
  process.exit(1);
}
